Adding raw track pT histo to JetHCorrelations, Adding corrected 2018 s.t.r.e. to JetHUtils

chughes90 opened the pull request:
Modified Files:

AliAnalysisTaskEmcalJetHCorrelations.h
AliAnalysisTaskEmcalJetHCorrelations.cxx

AliAnalysisTaskEmcalJetHUtils.h
AliAnalysisTaskEmcalJetHUtils.cxx

Description:

JetHCorrelations - Added an extra histogram for the raw pT track spectrum (so you don't have to project) - diagnostic

JetHUtils - Adding more accurate parameterization and parameters for 2018q s.t.r.e.

alibuild commented:
d4ca262ad2156e0c6e6b0accf30f5ca2c38606c6: approval required: 1 of @jgrosseo (Jan Fiete Grosse-Oetringhaus), @matplo (Mateusz Ploskon), @davidrohr (David Rohr), @adriansev (Adrian Sevcenco), @raymondEhlers (Raymond James Ehlers), @loizides (Constantinos Loizides), @vkucera (Vit Kucera), @shahor02 (Ruben Shahoyan), @mfasDa (Markus Fasel), @germainma (Marie Germain), @lcunquei (Leticia Cunqueiro Mendez), @qgp (Jochen Klein)

_Comment with `+1` to approve and allow automatic merging,or with `+test` to run tests only. **Please comment on the pull request: [click here](https://github.com/alisw/AliPhysics/pull/19667) and comment at the bottom of the page.**_

raymondEhlers commented on PWGJE/EMCALJetTasks/UserTasks/AliAnalysisTaskEmcalJetHUtils.h:
```diff
@@ -163,7 +171,7 @@ class FlowVectorCorrections {
   static void DefineHistograms(AliQnCorrectionsManager* QnManager, AliQnCorrectionsHistos* histos, TString histClass);
 };
-
+*/
```
Why are you commenting this out?

chughes90 replied:
Sorry for making this mistake again. I uncommented it. The reason I comment it out when testing locally is because if I do not, then I get the following errors when compiling it:

In file included from ./AliAnalysisTaskEmcalJetHPerformance.h:23:
/home/alidock/Raymond_Code/Raymond_Latest_Code/AliAnalysisTaskEmcalJetHUtils_v2.h:163:7: error: redefinition of 'FlowVectorCorrections'
class FlowVectorCorrections {
      ^
/home/alidock/alice/sw/slc7_x86-64/AliPhysics/master_ROOT6-local2/include/AliAnalysisTaskEmcalJetHUtils.h:155:7: note: previous definition is here
class FlowVectorCorrections {
 
...

fatal error: too many errors emitted, stopping now [-ferror-limit=]
Error in <TInterpreter::AutoParse>: Error parsing payload code for class PWGJE::EMCALJetTasks::AliAnalysisTaskEmcalJetHdEdxCorrelations with content:

raymondEhlers replied:
It's because you have two copies and didn't change the name of the class in your new copy

chughes90 commented on PWGJE/EMCALJetTasks/UserTasks/AliAnalysisTaskEmcalJetHUtils.cxx:
```diff
@@ -1138,8 +1135,9 @@ double AliAnalysisTaskEmcalJetHUtils::LHC18qrTrackingEfficiency(const double tra
-  return ((trackPt <= 3.5) * LHC18qrLowPtEfficiencyImpl(trackPt, params, 0) +
-      (trackPt > 3.5) * LHC18qrHighPtEfficiencyImpl(trackPt, params, 5));
+  return ((trackPt <= 2.7) * LHC18qrLowPtEfficiencyImpl(trackPt, params, 0) +
+      (trackPt > 2.7 && trackPt <= 10) * LHC18qrLowPtEfficiencyImpl(trackPt, params, 5) +
```
It is. Thank you for pointing this out, I missed it in local testing. Code still compiles and runs but it would have given another erroneous result.

alibuild commented:
904e37139fe66fb4260ef760456ef3999baeaeaa: approval required: 1 of @jgrosseo (Jan Fiete Grosse-Oetringhaus), @matplo (Mateusz Ploskon), @davidrohr (David Rohr), @adriansev (Adrian Sevcenco), @raymondEhlers (Raymond James Ehlers), @loizides (Constantinos Loizides), @vkucera (Vit Kucera), @shahor02 (Ruben Shahoyan), @mfasDa (Markus Fasel), @germainma (Marie Germain), @lcunquei (Leticia Cunqueiro Mendez), @qgp (Jochen Klein)

_Comment with `+1` to approve and allow automatic merging,or with `+test` to run tests only. **Please comment on the pull request: [click here](https://github.com/alisw/AliPhysics/pull/19667) and comment at the bottom of the page.**_

chughes90 commented:
Modified Files:

AliAnalysisTaskEmcalJetHUtils.h
AliAnalysisTaskEmcalJetHUtils.cxx

Description:

Uncommented `class FlowVectorCorrections` definition in JetHUtils.h. Corrected the pT efficiency for 2018 in JetHUtls.cxx to properly implement the "middle" piecewise efficiency (2.7 < pT < 10)

raymondEhlers commented:
+1

alibuild commented:
904e37139fe66fb4260ef760456ef3999baeaeaa: approved: will be automatically merged on successful tests
